scripts/linters/general_purpose_linter_test.py

```diff
@@ -143,615 +143,558 @@ class HTMLLintTests(test_utils.LinterTestBase):
             ['Line 30: In tests, please do not use browser.pause().'],
-            self.linter_stdout)
-        self.assert_failed_messages_count(self.linter_stdout, 1)
+            lint_task_report.messages)
+        self.assert_failed_messages_count(lint_task_report.all_messages, 1)
 
     def test_invalid_use_of_browser_sleep(self):
-        with self.print_swap:
-            general_purpose_linter.GeneralPurposeLinter(
-                [INVALID_SLEEP_FILEPATH], FILE_CACHE, True
-            ).perform_all_lint_checks()
+        linter = general_purpose_linter.GeneralPurposeLinter(
+            [INVALID_SLEEP_FILEPATH], FILE_CACHE)
+        lint_task_report = linter.check_bad_patterns()
         self.assert_same_list_elements(
             ['Line 30: In tests, please do not use browser.sleep().'],
-            self.linter_stdout)
-        self.assert_failed_messages_count(self.linter_stdout, 1)
+            lint_task_report.messages)
+        self.assert_failed_messages_count(lint_task_report.all_messages, 1)
 
     def test_invalid_use_of_browser_wait_for_angular(self):
-        with self.print_swap:
-            general_purpose_linter.GeneralPurposeLinter(
-                [INVALID_WAIT_FOR_ANGULAR_FILEPATH], FILE_CACHE, True
-            ).perform_all_lint_checks()
+        linter = general_purpose_linter.GeneralPurposeLinter(
+            [INVALID_WAIT_FOR_ANGULAR_FILEPATH], FILE_CACHE)
+        lint_task_report = linter.check_bad_patterns()
         self.assert_same_list_elements([
             'Line 30: In tests, please do not use '
-            'browser.waitForAngular().'], self.linter_stdout)
-        self.assert_failed_messages_count(self.linter_stdout, 1)
+            'browser.waitForAngular().'], lint_task_report.messages)
+        self.assert_failed_messages_count(lint_task_report.all_messages, 1)
 
     def test_invalid_use_of_fdescribe_ddescribe(self):
-        with self.print_swap:
-            general_purpose_linter.GeneralPurposeLinter(
-                [INVALID_FDESCRIBE_DDESCRIBE_FILEPATH], FILE_CACHE, True
-            ).perform_all_lint_checks()
+        linter = general_purpose_linter.GeneralPurposeLinter(
+            [INVALID_FDESCRIBE_DDESCRIBE_FILEPATH], FILE_CACHE)
+        lint_task_report = linter.check_bad_patterns()
         self.assert_same_list_elements([
             'Line 24: In tests, please use \'describe\' instead of '
-            '\'ddescribe\'or \'fdescribe\''], self.linter_stdout)
+            '\'ddescribe\'or \'fdescribe\''], lint_task_report.messages)
         self.assert_same_list_elements([
             'Line 28: In tests, please use \'describe\' instead of '
-            '\'ddescribe\'or \'fdescribe\''], self.linter_stdout)
-        self.assert_failed_messages_count(self.linter_stdout, 1)
+            '\'ddescribe\'or \'fdescribe\''], lint_task_report.messages)
+        self.assert_failed_messages_count(lint_task_report.all_messages, 1)
 
     def test_invalid_use_of_iit_fit(self):
```

Comment:
If someone adds new bad_pattern without any test the backend test will fail for coverage? If no, do you have any suggestions on how we can enforce having tests for such cases?